worker/src/RTC/RtpStreamSend.cpp

```diff
@@ -101,40 +114,34 @@ namespace RTC
 			this->startSeq = seq;
 		}
 
+		MS_ASSERT(
+		  this->buffer.size() <= MaxSeq,
+		  "StorageItemBuffer contains more than %" PRIu16 " entries",
+		  MaxSeq);
+
 		return true;
 	}
 
-	bool RtpStreamSend::StorageItemBuffer::RemoveFirst()
+	void RtpStreamSend::StorageItemBuffer::RemoveFirst()
 	{
-		return this->Remove(this->startSeq);
-	}
+		MS_ASSERT(!this->buffer.empty(), "buffer is empty");
 
-	bool RtpStreamSend::StorageItemBuffer::Remove(uint16_t seq)
-	{
-		if (this->buffer.empty())
-			return false;
+		auto storageItem = this->buffer[0];
 
-		auto idx{ static_cast<uint16_t>(seq - this->startSeq) };
+		// Reset (free RTP packet) the old storage item.
+		resetStorageItem(storageItem);
+		// Return into the pool.
+		StorageItemPool.Return(storageItem);
 
-		this->buffer[idx] = nullptr;
+		this->buffer[0] = nullptr;
 
-		// If we have erased the first element, remove all `nullptr` elements from the beginning of the buffer.
-		if (idx == 0)
-		{
-			while (!this->buffer.front())
-			{
-				this->buffer.pop_front();
-				this->startSeq++;
-			}
-		}
-		// If we have erased the last element, remove all `nullptr` elements from the end of the buffer.
-		else if (idx == static_cast<uint16_t>(this->buffer.size() - 1))
+		// Remove all `nullptr` elements from the beginning of the buffer.
+		// NOTE: Calling front on an empty container is undefined.
```

Comment:
Oh, wow, such an awkward API! I would have never thought that calling front can be so dangerous :facepalm: 

Can totally be a source of weird random issues I have seen.